fixtures/flight/server/region.js

```diff
@@ -45,25 +50,50 @@ app.get('/', async function (req, res) {
   );
   // const m = require('../src/App.js');
   const m = await import('../src/App.js');
-  const dist = process.env.NODE_ENV === 'development' ? 'dist' : 'build';
-  const data = await readFile(
-    path.resolve(__dirname, `../${dist}/react-client-manifest.json`),
-    'utf8'
-  );
+
+  let moduleMap;
+  let mainCSSChunks;
+  if (process.env.NODE_ENV === 'development') {
+    // Read the module map from the HMR server in development.
+    moduleMap = await (
+      await fetch('http://localhost:3000/react-client-manifest.json')
+    ).json();
+    mainCSSChunks = (
+      await (
+        await fetch('http://localhost:3000/entrypoint-manifest.json')
+      ).json()
+    ).main.css;
+  } else {
+    // Read the module map from the static build in production.
+    moduleMap = JSON.parse(
+      await readFile(
+        path.resolve(__dirname, `../build/react-client-manifest.json`),
+        'utf8'
+      )
+    );
+    mainCSSChunks = JSON.parse(
+      await readFile(
+        path.resolve(__dirname, `../build/entrypoint-manifest.json`),
+        'utf8'
+      )
+    ).main.css;
+  }
   const App = m.default.default || m.default;
-  res.setHeader('Access-Control-Allow-Origin', '*');
-  const moduleMap = JSON.parse(data);
-  const {pipe} = renderToPipeableStream(React.createElement(App), moduleMap);
+  const root = [
+    // Prepend the App's tree with stylesheets required for this entrypoint.
+    mainCSSChunks.map(filename =>
+      React.createElement('link', {
+        rel: 'stylesheet',
+        href: filename,
+        precedence: 'default',
+      })
+    ),
+    React.createElement(App),
```

Comment:
Why couldn't the `mainCSSChunks` be passed as prop to `App` which would then render the link elements in the head, similar to what is documented at https://beta.reactjs.org/reference/react-dom/server/renderToPipeableStream#reading-css-and-js-asset-paths-from-the-build-output?

Reply:
It couldn't but this shows that you don't have to - using a new feature that places it in the right place. So the App code can't mess it up by forgetting.

Reply:
Now I realise that this was a stupid question. I wasn't aware that this avoids the need to serialise them manually with `bootstrapScriptContent`, and instead "Float" is handling these elements automatically. Sorry for the noise my PR comments might create.